src/components/layout/sider.tsx

```diff
@@ -0,0 +1,402 @@
+    setOpen({ ...open, [key]: !open[key] })
+  }
+
+  const renderTreeView = (tree: ITreeMenu[], innerSelectedKey?: string) => {
+    return tree.map((item: ITreeMenu) => {
+      const { icon, label, route, name, children, parentName, options } = item
+      const { innerMeta } = item.meta;
+      const isOpen = open[item.key || ''] || false
+
+      const isSelected = item.key === innerSelectedKey
+      const isNested = !(pickNotDeprecated(innerMeta?.parent, options?.parent, parentName) === undefined)
+
+      if (children.length > 0) {
+        return (
+          <CanAccess
+            key={item.key}
+            resource={name.toLowerCase()}
+            action="list"
+            params={{
+              resource: item,
+            }}>
+            <div key={item.key}>
+              <Tooltip title={label ?? name} placement="right" disableHoverListener={!siderCollapsed} arrow>
+                <ListItemButton
+                  onClick={() => {
+                    if (siderCollapsed) {
+                      setSiderCollapsed(false)
+                      if (!isOpen) {
+                        handleClick(item.key || '')
+                      }
+                    } else {
+                      handleClick(item.key || '')
+                    }
+                  }}
+                  sx={{
+                    pl: isNested ? 4 : 2,
+                    justifyContent: 'center',
+                    marginTop: '8px',
+                  }}>
+                  <ListItemIcon
+                    sx={{
+                      justifyContent: 'center',
+                      minWidth: '24px',
+                      transition: 'margin-right 0.3s',
+                      marginRight: siderCollapsed ? '0px' : '12px',
+                      color: 'currentColor',
+                    }}>
+                    {icon ?? <ListOutlined />}
+                  </ListItemIcon>
+                  <ListItemText
+                    primary={label}
+                    primaryTypographyProps={{
+                      noWrap: true,
+                      fontSize: '14px',
+                    }}
+                  />
+                  {isOpen ? (
+                    <ExpandLess
+                      sx={{
+                        color: 'text.icon',
+                      }}
+                    />
+                  ) : (
+                    <ExpandMore
+                      sx={{
+                        color: 'text.icon',
+                      }}
+                    />
+                  )}
+                </ListItemButton>
+              </Tooltip>
+              {!siderCollapsed && (
+                <Collapse in={open[item.key || '']} timeout="auto" unmountOnExit>
+                  <List component="div" disablePadding>
+                    {renderTreeView(children, selectedKey)}
+                  </List>
+                </Collapse>
+              )}
+            </div>
+          </CanAccess>
+        )
```

Comment:
```suggestion
        ).filter(Boolean)
```
as some values of the array can be undefined (as the result of this condition `if (children.length > 0) {`)